src/OidcClient.ts

```diff
@@ -182,24 +183,45 @@ export class OidcClient {
             extraHeaders = { ...extraHeaders, "DPoP": dpopProof };
         }
 
-        await this._validator.validateSigninResponse(response, state, extraHeaders);
+        try {
+            await this._validator.validateSigninResponse(response, state, extraHeaders);
+        }
+        catch (err) {
+            if (err instanceof ErrorDPoPNonce) {
+                const dpopProof = await this.getDpopProof(this.settings.dpop!.store, err.nonce);
+                extraHeaders!["DPoP"] = dpopProof;
```

Comment:
Why not doing this before hand? I dislike doing `validateSigninResponse` twice...

Reply:
Apologies, perhaps I wasn't Clea enough in the PR description: that's how it works according to the spec. 

The nonce is supplied by the server, and the mechanism is through a bad request response from the token endpoint when fetching a token. Example from the [spec](https://datatracker.ietf.org/doc/html/rfc9449#name-authorization-server-provid):

```
For example, in response to a token request without a nonce when the authorization server requires one, the authorization server can respond with a DPoP-Nonce value such as the following to provide a nonce value to include in the DPoP proof
HTTP/1.1 400 Bad Request
 DPoP-Nonce: eyJ7S_zG.eyJH0-Z.HX4w-7v

 {
  "error": "use_dpop_nonce",
  "error_description":
    "Authorization server requires nonce in DPoP proof"
 }
[Figure 20](https://datatracker.ietf.org/doc/html/rfc9449#figure-20): [HTTP 400 Response to a Token Request without a Nonce](https://datatracker.ietf.org/doc/html/rfc9449#name-http-400-response-to-a-toke)
Other HTTP headers and JSON fields MAY also be included in the error response, but there MUST NOT be more than one DPoP-Nonce header.

Upon receiving the nonce, the client is expected to retry its token request using a DPoP proof including the supplied nonce value in the nonce claim of the DPoP proof. An example unencoded JWT payload of such a DPoP proof including a nonce is shown below.

 {
  "jti": "-BwC3ESc6acc2lTc",
  "htm": "POST",
  "htu": "https://server.example.com/token",
  "iat": 1562262616,
  "nonce": "eyJ7S_zG.eyJH0-Z.HX4w-7v"
 }
```

If we already have [the nonce stored,](https://github.com/authts/oidc-client-ts/pull/1569/files#diff-c833a002f3fa487666ed42d9742ea5ae2fa4ada00227943616583ed0c3204120R202-R209) we will not make the request twice unless the server responds that the nonce is invalid.

Reply:
Understood. Thanks for clarification.

For future reads of the code it is very helpful to have the right spec section documented near the affected code. Maybe you can add the link you provided above as a comment before the re-try blocks...

Reply:
No problem, that is a fair comment and a good idea 👍